libass/ass_font.c

```diff
@@ -515,6 +516,12 @@ void ass_face_set_size(FT_Face face, double size)
     FT_Request_Size(face, &rq);
 }
 
+bool ass_face_is_postscript(FT_Face face)
+{
+    PS_FontInfoRec postscript_info;
+    return !FT_Get_PS_Font_Info(face, &postscript_info);
```

Comment:
i’m trusting you to have verified FreeType doesn’t synthesise values here for pure TrueType fonts, though it would probably good to add regrioons test samples to libass-tests to ensure this doesn't change in the future without us noticing

FreeType 2.1.2’s changelog also [mentions](https://gitlab.freedesktop.org/freetype/freetype/-/blob/b875924a6fc176d46fc9f82c616db631fc4d665c/docs/CHANGES#L3899) the function now working with CID and Type42 fonts. The latter appear to be a compatibility wrapper around TrueType fonts iiuc, are we sure GDI treats them as PostScript outline fonts?

Reply:
This is just our existing implementation moved from `ass_fontselect`. It has worked thus far.

Reply:
As for Type42 fonts, I’ve never seen one and I don’t know what they look like; if you have an example, I’m happy to test it on Windows. Our tests until now have been with TrueType fonts with or without OpenType extensions; OpenType/CFF fonts (not sure if we’ve explicitly tested fonts using the `CFF2` table, but I assume they should behave the same as those using `CFF`); and Type 1 fonts (which enjoy limited support on some systems and are completely unsupported on others).

Reply:
I converted DejaVu Sans to a Type42 font using `ttftotype42` from [LCDF Typetools](https://www.lcdf.org/type/) *(conversion of the Jester font from the original issue failed with checksum errors)*. In libass with this patch they do indeed get different slant values using `mpv --sub-font-provider=none --sub-fonts-dir=/tmp/empty`: [type42_sample.zip](https://github.com/libass/libass/files/15283911/type42_sample.zip)

Reply:
I can’t tell whether VSFilter uses this font because I have a system font of the same name, but extracting the attachment and trying to open it produces an error (“not a font file”). I strongly suspect this format isn’t supported at all.

Reply:
verified VSFilter/GDI doesn't use such files on IRC using a renamed font sample